### PR TITLE
Double check IW is the notebook changing when handling scroll events

### LINKED ISSUE #155092: Interactive window needs to check notebook URI from execution state service
https://github.com/microsoft/vscode/blob/a1f2dae7b6fce98d68c7393bf64ceb632e98df59/src/vs/workbench/contrib/interactive/browser/interactiveEditor.ts#L151-L153

I just randomly noticed this - the notebookExecutionStateService is a global service and this event will be fired for any notebook. The event object has a notebook URI on it, and this handler needs to check that the event is fired for this notebook, otherwise executions in other notebooks with a cell handle overlap could affect this interactive window

### PR DESCRIPTION
**Pull request by rchiodo**
This PR fixes #155092

### CODE REVIEW

File: src/vs/workbench/contrib/interactive/browser/interactiveEditor.ts [hunk 1]
```diff
@@ -152,9 +152,11 @@ export class InteractiveEditor extends EditorPane {
 		codeEditorService.registerDecorationType('interactive-decoration', DECORATION_KEY, {});
 		this._register(this.#keybindingService.onDidUpdateKeybindings(this.#updateInputDecoration, this));
 		this._register(this.#notebookExecutionStateService.onDidChangeCellExecution((e) => {
-			const cell = this.#notebookWidget.value?.getCellByHandle(e.cellHandle);
-			if (cell && e.changed?.state) {
-				this.#scrollIfNecessary(cell);
+			if (e.notebook.toString() === this.#notebookWidget.value?.viewModel?.notebookDocument.uri.toString()) {
```
Comment (rebornix) [inline src/vs/workbench/contrib/interactive/browser/interactiveEditor.ts:155]: It's safe touse `isEqual` from `vs/base/common/resources` here to compare the notebook Uri.

Comment (rchiodo) [reply src/vs/workbench/contrib/interactive/browser/interactiveEditor.ts:155]: Will change. Thanks.

**Review by rebornix — APPROVED** [review]
👏

File: src/vs/workbench/contrib/interactive/browser/interactiveEditor.ts [hunk 2]
```diff
@@ -152,9 +153,11 @@ export class InteractiveEditor extends EditorPane {
 		codeEditorService.registerDecorationType('interactive-decoration', DECORATION_KEY, {});
 		this._register(this.#keybindingService.onDidUpdateKeybindings(this.#updateInputDecoration, this));
 		this._register(this.#notebookExecutionStateService.onDidChangeCellExecution((e) => {
-			const cell = this.#notebookWidget.value?.getCellByHandle(e.cellHandle);
-			if (cell && e.changed?.state) {
-				this.#scrollIfNecessary(cell);
+			if (isEqual(e.notebook, this.#notebookWidget.value?.viewModel?.notebookDocument.uri)) {
```
Comment (roblourens) [inline src/vs/workbench/contrib/interactive/browser/interactiveEditor.ts:156]: The event also has a helper method `affectsNotebook` that does the same as this

**Review by roblourens — APPROVED** [review]

**Pull request merged**